js_src/_dart_helpers.js

```diff
@@ -103,13 +103,26 @@ function _createReactDartComponentClass2(dartInteropStatics, componentStatics, j
     componentWillUnmount() {
       dartInteropStatics.handleComponentWillUnmount(this.dartComponent);
     }
+    componentDidCatch(error, info) {
+      dartInteropStatics.handleComponentDidCatch(this.dartComponent, error, info);
+    }
+    static getDerivedStateFromError(error) {
+      return dartInteropStatics.handleGetDerivedStateFromError(componentStatics, error);
+    }
     render() {
       var result = dartInteropStatics.handleRender(this.dartComponent, this.props, this.state, this.context);
       if (typeof result === 'undefined') result = null;
       return result;
     }
   }
 
+  // Delete methods that the user does not want to include (such as error boundary event).
+  skipMethods.forEach((method) => {
+    if (ReactDartComponent2[method] !== undefined) {
+      delete ReactDartComponent2[method];
```

Comment:
Does deletion work for non-static methods?

I would think you'd have to do:
```js
delete ReactDartComponent2.prototype[method];
```

Reply:
Could we also add some tests that this is working properly?
- lifecycle methods are not called for skipped static methods
- lifecycle methods are not called for skipped instance methods

Reply:
> ```js
> delete ReactDartComponent2.prototype[method];
> ```

Good call! Adding tests now.